envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/NodeMetadata.kt

```diff
@@ -268,6 +269,46 @@ private fun Value.toOutgoingTimeoutPolicy(default: Outgoing.TimeoutPolicy): Outg
     return Outgoing.TimeoutPolicy(idleTimeout ?: default.idleTimeout, requestTimeout ?: default.requestTimeout)
 }
 
+private fun Value.toOAuth(properties: SnapshotProperties): OAuth {
+    val provider = this.field("provider").toOauthProvider(properties)
+    val policy = this.field("policy").toOAuthPolicy(properties.jwt.defaultOAuthPolicy)
+    val verification = this.field("verification").toOAuthVerification(properties.jwt.defaultVerificationType)
+
+    return OAuth(provider, verification, policy)
+}
+
+fun Value?.toOauthProvider(properties: SnapshotProperties) = this?.stringValue
+    ?.takeIf { it.isNotEmpty() }
+    ?.let { if (properties.jwt.providers.keys.contains(it)) {
+            it
+        } else {
+            throw NodeMetadataValidationException("Invalid OAuth provider value: $it")
+        }
+    }
+    ?: throw NodeMetadataValidationException("OAuth provider value cannot be null")
+
+fun Value?.toOAuthVerification(defaultVerification: OAuth.Verification) = this?.stringValue
+    ?.takeIf { it.isNotEmpty() }
+    ?.let {
+        when (it) {
+            "offline" -> OAuth.Verification.OFFLINE
+            else -> throw NodeMetadataValidationException("Invalid OAuth verification value: $it")
+        }
+     }
+    ?: defaultVerification
+
+fun Value?.toOAuthPolicy(defaultPolicy: OAuth.Policy) = this?.stringValue
+    ?.takeIf { it.isNotEmpty() }
+    ?.let {
+        when (it) {
+            "allow_missing" -> OAuth.Policy.ALLOW_MISSING
```

Comment:
maybe Policy names should be in camel case to be consistent with other incomming permisions properties?